eMarket/app/urls.py

```diff
@@ -9,7 +9,10 @@
     path('registration_success/', views.registration_success, name='registration_success'),
     path('add_product/', views.add_product, name='add_product'),
     path('shopping_cart/', views.shopping_cart, name='shopping_cart'),
-    path('payment_confirmed/', views.payment_confirmed, name='payment_confirmed'),
+    path('add_to_cart/<product_id>/', views.add_to_cart, name = 'add_to_cart'),
+    path('remove_from_cart/<product_id>/', views.remove_from_cart, name = 'remove_from_cart'),
+    path('checkout/', views.checkout, name = 'checkout'),
+    path('payment_confirmation/', views.payment_confirmed, name='payment_confirmation'),
```

Comment:
It's typically best practice to keep url and file naming consistent but it's not required here. ie. path('payment_confirmed/', views.payment_confirmed, name='payment_confirmed'),